InteractionBase/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -40,18 +41,42 @@ bool CAddNodeToViewByName::canInterpret(Visualization::Item*, Visualization::Ite
 	return commandTokens.size() >= 1 && commandTokens.first() == name();
 }
 
-CommandResult* CAddNodeToViewByName::execute(Visualization::Item*, Visualization::Item* target,
-		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>&)
+CommandResult* CAddNodeToViewByName::execute(Visualization::Item* source, Visualization::Item* target,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& cursor)
 {
 	if (commandTokens.size() < 2)
 		return new CommandResult(new CommandError("Please specify a node to add"));
 
+	auto currentView = target->scene()->currentViewItem();
+	QPoint posToInsert;
+	if (source == currentView && isHorizontal(cursor->region()))
```

Comment:
Making a distinction between cursor types, only makes sense, if the cursor is currently directly in the ViewItem. If the cursor is currently anywhere in a child of the ViewItem, the user would find it very odd that new views are sometimes created to the left and sometimes below.

So, when the cursor's `owner` is the view item, you should make a distinction based on where the cursor is. However, don't do it with hacky methods. You can cast that cursor to `LayoutCursor` and use its `x()`, `y()` and `index()` methods to directly find the current position.

When the cursor's `owner` is not a viewitem, then please do not distinguish between vertical and horizontal cursors.